api/src/main/java/io/opentelemetry/metrics/DoubleCounter.java

```diff
@@ -64,6 +64,16 @@
    */
   void add(double increment, Labels labels);
 
+  /**
+   * Adds the given {@code increment} to the current value. The values cannot be negative.
+   *
+   * <p>The value added is associated with the current {@code Context} and with empty labels.
+   *
+   * @param increment the value to add.
+   * @since 0.8.0
+   */
+  void add(double increment);
```

Comment:
Not related to this PR at all - @jkwatson I was surprised to see here that we don't allow negative additions

Reply:
The UpDownCounters are the ones that allow negative increments.

Reply:
Ah got it! I think... do we need two types of counters? Just food for thought no strong opinion, mostly just used to things like `AtomicInteger` in terms of API.

Reply:
There was intense debate in the metric specs about this. The rationale (for good or ill) was that having different instruments that reflect the different semantics of the counting was valuable for instrumentation, to make it very clear what sort of counting you were doing. We went the opposite way from OpenCensus, which (I think?) has one instrument for everything, and lots of options.